osquery/events/linux/auditeventpublisher.cpp

```diff
@@ -129,13 +134,122 @@ void AuditEventPublisher::ProcessEvents(
 
       // SELinux events
     } else if (selinux_event_set.find(audit_event_record.type) !=
-               selinux_event_set.end()) {
+                   selinux_event_set.end() &&
+               audit_event_record.fields.find("apparmor") ==
+                   audit_event_record.fields.end()) {
       AuditEvent audit_event;
       audit_event.type = AuditEvent::Type::SELinux;
       audit_event.record_list.push_back(audit_event_record);
 
       event_context->audit_events.push_back(audit_event);
 
+    } else if (apparmor_event_set.find(audit_event_record.type) !=
+                   apparmor_event_set.end() &&
+               audit_event_record.fields.find("apparmor") !=
```

Comment:
Is this check required? Is there anyway to restructure these if/else if checks to be more explicit for SELinux/AppArmor? There might not be, but the way this reads off to me makes me think in the future, if folks have additional tables they'd like to extract out of audit information, this list of conditional checks will continue to grow in a bad way. If the `audit_event_record.type` is inside of our `event_set`, and we've already assured it's not an `selinux` event, do we need to ensure that the fields contain `"apparmor"`? Will it always be there? I wonder if there isn't a way to restructure some of these data structures to allow for neater event type checking, curious what others think here.

Reply:
Well, I also don't really like this sort of `if/else` mess, but as I explained above, AppArmor uses only type=1400 code and this code is shared with SELinux, so there only way to tell AppArmor events from SELinux ones is to check for the `apparmor=` substring. 

I'll try to come up with something for this place.